src/main/java/bio/terra/cli/context/Workspace.java

```diff
@@ -0,0 +1,241 @@
+
+  // properties of the workspace
+  public final UUID id;
+  public final String name; // not unique
+  public final String description;
+  public final String googleProjectId;
+
+  // name of the server where this workspace exists
+  public final String serverName;
+
+  // email of the user that loaded the workspace to this machine
+  public final String terraUserEmail;
+
+  // list of resources (controlled & referenced)
+  private List<Resource> resources;
+
+  @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+  private Workspace(
+      @JsonProperty("id") UUID id,
+      @JsonProperty("name") String name,
+      @JsonProperty("description") String description,
+      @JsonProperty("googleProjectId") String googleProjectId,
+      @JsonProperty("serverName") String serverName,
+      @JsonProperty("terraUserEmail") String terraUserEmail,
+      @JsonProperty("resources") List<Resource> resources) {
+    this.id = id;
+    this.name = name;
+    this.description = description;
+    this.googleProjectId = googleProjectId;
+    this.serverName = serverName;
+    this.terraUserEmail = terraUserEmail;
+    this.resources = resources;
+  }
+
+  /**
+   * Create a new workspace and set it as the current workspace.
+   *
+   * @param name optional display name
+   * @param description optional description
+   */
+  public static Workspace create(String name, String description) {
+    // call WSM to create the workspace object and backing Google context
+    WorkspaceDescription createdWorkspace =
+        new WorkspaceManagerService().createWorkspace(name, description);
+    logger.info("Created workspace: {}", createdWorkspace);
+
+    // convert the WSM object to a CLI object
+    Workspace workspace = fromWSMObject(createdWorkspace);
+
+    // fetch the pet SA credentials for the user + this workspace
+    GlobalContext globalContext = GlobalContext.get();
+    globalContext.requireCurrentTerraUser().fetchPetSaCredentials();
+
+    // update the global context with the current workspace
+    GlobalContext.get().setCurrentWorkspace(workspace);
+    return workspace;
+  }
+
+  /**
+   * Load an existing workspace and set it as the current workspace.
+   *
+   * @param id workspace id
+   */
+  public static Workspace load(UUID id) {
+    // call WSM to fetch the existing workspace object and backing Google context
+    WorkspaceDescription loadedWorkspace = new WorkspaceManagerService().getWorkspace(id);
+    logger.info("Loaded workspace: {}", loadedWorkspace);
+
+    // convert the WSM object to a CLI object
+    Workspace workspace = fromWSMObject(loadedWorkspace);
+
+    // fetch the pet SA credentials for the user + this workspace
+    GlobalContext globalContext = GlobalContext.get();
+    globalContext.requireCurrentTerraUser().fetchPetSaCredentials();
+
+    // update the global context with the current workspace
+    GlobalContext.get().setCurrentWorkspace(workspace);
```

Comment:
I think this is a "initializing" a Workspace for the CLI code that's shared in `create` and `load`. I think we should have a shared private function for it, or maybe even move it out of the `Workspace` class, though I'm not sure where to put it.

Reply:
Right, yes. I considered putting it in a separate function, but decided against because it's only 3 lines. If it gets longer, agreed on factoring it out.